db/model/archival/archivalDb.ts

```diff
@@ -792,124 +879,306 @@ export const getPostChecksumsFromDb = async (
+        (slug) => !slugsRedirectingToMultiDim.has(slug)
+    )
     const grapherChecksums = await getGrapherChecksumsFromDb(
         knex,
-        Array.from(resolvedGrapherSlugs)
+        slugsForChartLookup
     )
     const grapherChecksumsMap = new Map(
         grapherChecksums.map((c) => [c.chartSlug, c])
     )
 
-    const multiDimSlugs = Array.from(resolvedGrapherSlugs).filter(
-        (slug) => !grapherChecksumsMap.has(slug)
+    // Step 5: Classify each resolved slug and fetch multi-dim checksums
+    const grapherResolutions = classifyGrapherSlugs(
+        Array.from(slugsAfterChartRedirects),
+        grapherChecksumsMap,
+        multiDimRedirects
     )
+
+    // Collect all multi-dim slugs (both from redirects and unresolved graphers)
+    const multiDimSlugs = new Set<string>()
+    for (const resolved of grapherResolutions.values()) {
+        if (resolved.type === "multiDim") {
+            multiDimSlugs.add(resolved.slug)
+        }
+    }
+
     const multiDimChecksums = await getMultiDimChecksumsFromDb(
         knex,
-        multiDimSlugs
+        Array.from(multiDimSlugs)
     )
     const multiDimChecksumsMap = new Map(
         multiDimChecksums.map((m) => [m.multiDimSlug, m])
     )
 
-    const explorerChecksums = await getExplorerChecksumsFromDb(
+    // Build resolution map from original slugs to resolved content
+    const finalResolutions = new Map<string, ResolvedSlug>()
+    for (const originalSlug of originalSlugs) {
+        const afterChartRedirect =
+            chartRedirects.get(originalSlug) ?? originalSlug
+        const resolved = grapherResolutions.get(afterChartRedirect)
+        if (resolved) {
+            finalResolutions.set(originalSlug, resolved)
+        }
+    }
+
+    return {
+        grapherChecksumsMap,
+        multiDimChecksumsMap,
+        grapherResolutions: finalResolutions,
+    }
+}
+
+/**
+ * Resolves explorer slugs, checking for multi-dim redirects.
+ * Returns checksums and adds any new multi-dims to the provided map.
+ */
+async function resolveExplorerSlugsAndFetchChecksums(
+    knex: db.KnexReadonlyTransaction,
+    originalSlugs: string[],
+    existingMultiDimMap: Map<string, MultiDimChecksumsObjectWithHash>
+): Promise<{
+    explorerChecksumsMap: Map<string, ExplorerChecksumsObjectWithHash>
+    explorerResolutions: Map<string, ResolvedSlug>
+}> {
+    if (originalSlugs.length === 0) {
+        return {
+            explorerChecksumsMap: new Map(),
+            explorerResolutions: new Map(),
+        }
+    }
+
+    // Check for multi-dim redirects
+    const multiDimRedirects = await resolveMultiDimRedirects(
         knex,
-        Array.from(allExplorerSlugs)
+        originalSlugs,
+        "/explorers/"
     )
-    const explorerChecksumsMap = new Map(
-        explorerChecksums.map((e) => [e.explorerSlug, e])
+
+    const slugsRedirectingToMultiDim = new Set(
+        originalSlugs.filter((slug) => multiDimRedirects.get(slug) !== slug)
     )
```

Comment:
I guess this would also filter out the case when `/explorers/covid` redirects to `/grapher/covid`?